File: Logic-2/roundSum/RoundSum.java
package roundSum;

public class RoundSum {
    //Dados 3 números inteiros ("a","b" e "c"), retorne a sua soma obedecendo a seguinte lógica:
    //-Se o dígito mais a direita for 5 ou mais, arredonde-o para o múltipo de 10 mais a cima. Ex.: 15 -> 20.
    //-Se o dígito mais a direita for menor que 5, arredonde-o para o múltipo de 10 mais a baixo. Ex.: 14 -> 10.
    //Para isso crie uma função auxiliar "public int round10(int num)" que retornará o número ajustado a essa regra.

    public int roundSum(int a, int b, int c) {
        return round10(a) + round10(b) + round10(c);
    }

    public int round10(int num){
        int unidade = num % 10;
        if (unidade >= 5) return num + (10-unidade);
        else return num - unidade;
    }

    //LEVEMENTE DIFERENTE, MAS INTERESSANTE
    public int roundSum_2(int a, int b, int c) {
        return round10(a) + round10(b) + round10(c);
    }

    // Helper function to round a value based on its last digit.
    public int round10_2(int num) {
        int remainder = num % 10;
        num -= remainder;
        if (remainder >= 5) {
            num += 10;
        }
        return num;
    }
}

File: Python/Logic-2/round_sum.py
"""
Dado 3 números inteiros ("a", "b" e "c"), faça os devidos arredondamentos para o
valor mais próximo de um múltiplo de 10 (caso o dígito mais a direta for 5 ou mais,
arredonde para cima, do contrário para baixo) e retorne a sua soma. Para isso
crie um método auxiliar "def round10(num):" que será chamado 3 vezes e retornará
o valor adequado para a regra.
"""


def round_sum(a, b, c):
    soma = 0
    soma += round10(a)
    soma += round10(b)
    soma += round10(c)
    return soma


def round10(num):
    return (int(num / 10) + 1) * 10 if num % 10 >= 5 else int(num / 10) * 10


##############
# OUTRO JEITO DE SE FAZER
def round_sum_2(a, b, c):
    return round10(a) + round10(b) + round10(c)


def round10_2(num):
    mod = num % 10
    num -= mod
    if mod >= 5:
        num += 10
    return num

##############


print(round_sum(12, 15, 2))
print(round_sum_2(32, 25, 4))
